client/signup/config/flows.js

```diff
@@ -63,11 +63,15 @@ function getSiteDestination( dependencies ) {
 }
 
 function getRedirectDestination( dependencies ) {
-	if (
-		dependencies.oauth2_redirect &&
-		dependencies.oauth2_redirect.startsWith( 'https://public-api.wordpress.com' )
-	) {
-		return dependencies.oauth2_redirect;
+	try {
+		if (
+			dependencies.oauth2_redirect &&
+			new URL( dependencies.oauth2_redirect ).host === 'public-api.wordpress.com'
```

Comment:
Might want to be a bit more circumspect here. `new URL(‘foo’)` or a similar malformed URL can result in a type error rather than an instance of URL. We had to revert a similar usage on an earlier series of PRs refactoring `lib/url` and friends. Unsure of the likelihood of a bad `oauth2_redirect` value, but probably worth guarding against.

Reply:
Won't the `try / catch` block that we already have guard against those kinds of errors, too?